### PR TITLE
updated libraries folder

### PR DESCRIPTION
**Pull request by IFX-Anusha**
By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

CONTRIBUTING.md also tells you what to expect in the PR process.

Description
Provide the information we need to review your PR. What problem does the pull request solve? "Bug fix" is not a good description.

Related Issue
If you opened an issue before creating the PR, point to it here.

Context
What do we need to know about your development environment, tools, target, and so on. Screenshots are always helpful if there is a UI element to this PR.

### CODE REVIEW

**Review by techpaul — CHANGES_REQUESTED** [review]
This appears to me a MASS deletion of libraries YOU have NOT used

Many are for benefit of others and one you have deleted would make YOUR examples board AGNOSTIC, namely LED, which means an example can use LED.Add( pin) and LED.On or LED.Off or LED.Toggle to ALWAYS work on any board whatever method of On being 1 or 0.

This is determined by define XMC_LED_ON in pins_arduino.h
This may break other examples and libraries.

Highlly suspicious of how this PR has been done

**Review by techpaul — CHANGES_REQUESTED** [review]
Most RTC functions are supported on XMC1 series AS WELL

Hibernate support was dropped three years ago as Arduino framework does not support detection of Reset from hibernate and staorage to restore system state (including ALL buffers)

File: libraries/SPI/src/utility/xmc_spi_conf.c [hunk 1]
```diff
@@ -107,46 +107,6 @@ XMC_SPI_t XMC_SPI_0 =
     }
 };
 
-#elif defined(XMC1400_Arduino_Kit)
```
Comment (jaenrig-ifx) [inline libraries/SPI/src/utility/xmc_spi_conf.c:110]: Mmm ... the fork has not been updated to the latest develop-2.x branch. And the board is removed from the built-in library.  I believe this is one of the boards to keep in 2.x.

File: libraries/SPI/src/utility/xmc_spi_conf.h [hunk 2]
```diff
@@ -60,10 +60,6 @@ extern XMC_SPI_t XMC_SPI_0;
 #define NUM_SPI  			1
 extern XMC_SPI_t XMC_SPI_0;
 
-#elif defined(XMC1400_Arduino_Kit)
```
Comment (jaenrig-ifx) [inline libraries/SPI/src/utility/xmc_spi_conf.h:63]: Same as [xmc_spi_conf.c](https://github.com/Infineon/XMC-for-Arduino/pull/195/files#diff-f0a5b6e4a58530712f1cbce9463eed9daedae46d0f0acb1f8fde3641bd66ed4c)

File: libraries/Wire/src/utility/xmc_i2c_conf.c [hunk 3]
```diff
@@ -197,40 +197,6 @@ XMC_I2C_t XMC_I2C_0 =
     .protocol_irq_service_request       	  = 5
 };
 
-#elif defined(XMC1400_Arduino_Kit)
```
Comment (jaenrig-ifx) [inline libraries/Wire/src/utility/xmc_i2c_conf.c:200]: Same as [xmc_spi_conf.c](https://github.com/Infineon/XMC-for-Arduino/pull/195/files#diff-f0a5b6e4a58530712f1cbce9463eed9daedae46d0f0acb1f8fde3641bd66ed4c)

File: libraries/Wire/src/utility/xmc_i2c_conf.h [hunk 4]
```diff
@@ -64,10 +64,6 @@ extern XMC_I2C_t XMC_I2C_1;
 #define NUM_I2C  1
 extern XMC_I2C_t XMC_I2C_0;
 
-#elif defined(XMC1400_Arduino_Kit)
```
Comment (jaenrig-ifx) [inline libraries/Wire/src/utility/xmc_i2c_conf.h:67]: Same as [xmc_spi_conf.c](https://github.com/Infineon/XMC-for-Arduino/pull/195/files#diff-f0a5b6e4a58530712f1cbce9463eed9daedae46d0f0acb1f8fde3641bd66ed4c)

File: libraries/SPI/src/utility/xmc_spi_conf.c (hunk 1 above)
Comment (OlafFilies) [reply libraries/SPI/src/utility/xmc_spi_conf.c:110]: Yes it should stay as the XMC1400 will be the successor or the XMC1100

**Comment by jaenrig-ifx** [comment]
@techpaul this develop-2.x is a minimal implementation of xmc-for-arduino.
The current develop/master (v1.x) keeps all the built-in libs.

Only some relevant boards and built-in libraries building and functionally evaluated will be kept. 
Many of them will be added later (again) as they are functionally verified and provided testing examples which can be automated and skipped specific hardware dependencies like external sensors, etc.

File: libraries/SPI/src/utility/xmc_spi_conf.c (hunk 1 above)
Comment (IFX-Anusha) [reply libraries/SPI/src/utility/xmc_spi_conf.c:110]: Yes. Sorry. I missed out on the updates inside the libraries folder. Will update that

**Comment by techpaul** [comment]
@jaenrig-ifx 
> @techpaul this develop-2.x is a minimal implementation of xmc-for-arduino. The current develop/master (v1.x) keeps all the built-in libs.

Lack of overall description and any details, sounds like an internal test that should have been a separate internal FORK first.

> Only some relevant boards and built-in libraries building and functionally evaluated will be kept. Many of them will be added later (again) as they are functionally verified and provided testing examples which can be automated and skipped specific hardware dependencies like external sensors, etc.

So so automated testing and inability to catch the mistake in the same device SPI test, with no external sensors

**Comment by techpaul** [comment]
A better rework would be to move the xmc_pi_conf and xmc_i2c_conf files to variants folder, for easier maintainability. Note this makes NO difference to compilation paths if in same folder as pins_arduino.h

File: libraries/RTC/examples/HibernateRTCAlarm/HibernateRTCAlarm.ino [hunk 5]
```diff
@@ -0,0 +1,62 @@
+/*
```
Comment (techpaul) [inline libraries/RTC/examples/HibernateRTCAlarm/HibernateRTCAlarm.ino:1]: Let me explain this another way, this example has been deleted from relase version of XMC-for-Arduino something like 3 years ago.

The person who wrote this example did not understand how Hibernate works, when the CPU comes out of hibernate the CPU is RESET apart from hibernate falags. This means the code will run Setup and Loop again

The function alarmMatch() will NEVER be reached.

xmc.enableHibernate was removed from DeviceControl library ages ago this example should be removed.

**Review by techpaul — CHANGES_REQUESTED** [review]
Remove functionality for Hiberante examples

File: libraries/RTC/examples/HibernateRTCAlarm/HibernateRTCAlarm.ino (hunk 5 above)
Comment (techpaul) [reply libraries/RTC/examples/HibernateRTCAlarm/HibernateRTCAlarm.ino]: This suggests your base for develop-2.x was based on a VERY OLD version of the package

Comment (jaenrig-ifx) [reply libraries/RTC/examples/HibernateRTCAlarm/HibernateRTCAlarm.ino]: > A better rework would be to move the xmc_pi_conf and xmc_i2c_conf files to variants folder, for easier maintainability. Note this makes NO difference to compilation paths if in same folder as pins_arduino.h

Yes, that will be considering in an upcoming improvement 👍

**Comment by jaenrig-ifx** [comment]
Together with the upcoming board clean up 👍

**Review by jaenrig-ifx — APPROVED** [review]

**Review by OlafFilies — APPROVED** [review]

**Pull request merged**